### ISSUE TITLE
Chaining functions accepting callbacks & `*args` typed with `TypeVarTuple` fails to bind types

### ISSUE THREAD

**Issue by Enegg** [microsoft/pyright#7140]
**Describe the bug**
Followup to #7136
Pyright reports an error while passing a function A, accepting another function B and positional args to pass to it, to a similarly working function C in the fashion of `C(A, B, *args_to_b)`. It is related to the resolution of `TypeVarTuple`s from function signature.

**Code or Screenshots**
```python
import anyio  # 4.2.0

async def foo() -> int:
    def runner() -> None:
        ...

    async with anyio.create_task_group() as tg:
        tg.start_soon(anyio.to_thread.run_sync, runner)  # error here
```
![image](https://github.com/microsoft/pyright/assets/42005170/b940cf6d-24a0-42be-9dd4-48d7e424d133)

<details>
<summary>Signatures of called functions</summary>

![image](https://github.com/microsoft/pyright/assets/42005170/083ddf39-2e8b-4370-8183-54f0f975ba5a)
![image](https://github.com/microsoft/pyright/assets/42005170/468bfca5-f51b-42f1-8239-c42cdf3283d0)

</details>

**VS Code extension or command-line**
VS Code extension - Pylance `v2024.1.101` (prerelease)

**Comment by erictraut** [comment]
Thanks for the bug report. I agree this should type check without errors.

For my own benefit, here's a simplified self-contained repro:

```python
from typing import Any, Awaitable, Callable

async def run_sync[*Ts, R](func: Callable[[*Ts], R], *args: *Ts) -> R:
    return func(*args)

def runner() -> None: ...

def start_soon[*Xs](func: Callable[[*Xs], Awaitable[Any]], *args: *Xs) -> None:
    pass

start_soon(run_sync, runner)
```

I'll investigate further.

**Comment by FynnBe** [comment]
I think I stumbled over the same issue, so here goes one more example/test case:

```python
from typing import Callable, Tuple

from typing_extensions import TypeVarTuple, Unpack

Ts = TypeVarTuple("Ts")


def foo(*tup: Unpack[Ts]) -> Tuple[Unpack[Ts]]:
    return tup


def bar(f: Callable[[Unpack[Ts]], Tuple[Unpack[Ts]]]):
    result = f(1, 2, 3)
    # Type "*tuple[Literal[1], Literal[2], Literal[3]]" cannot be assigned to type "*Ts@bar"

    print(result)


_ = foo(1, 2, 3)
bar(foo)

```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.363, which I just published.
